datafusion/sqllogictest/test_files/csv_files.slt

```diff
@@ -336,3 +336,23 @@ id message
 05)good test
 4 unquoted
 value end
+
+statement ok
+CREATE EXTERNAL TABLE stored_table_with_cr_terminator (
+col1 TEXT,
+col2 TEXT
+) STORED AS CSV
+LOCATION '../core/tests/data/cr_terminator.csv'
+OPTIONS ('format.terminator' E'\r', 'format.has_header' 'true');
+
+# TODO: It should be passed but got the error: External error: query failed: DataFusion error: Object Store error: Generic LocalFileSystem error: Requested range was invalid
+# query TT
+# select * from stored_table_with_cr_terminator;
```

Comment:
I still got this error in the sqllogictests but I don't know why. The same case in [datafusion/core/src/datasource/physical_plan/csv.rs](https://github.com/apache/datafusion/pull/12263/files#diff-264e8b6a69a6d6132ac5bf88de1e27a8539b6840619c4d51d2c0c8426cc79612) works fine. I tried to find if there are any special configs for object_store in the sqllogictests but I found nothing. 🤔

Reply:
I bet it is a bug and not related to this PR

What I think we should do is file a ticket tracking the issue and then merge this PR and work on figuring it out as a follow on.

Reply:
I filed https://github.com/apache/datafusion/issues/12328 to trace this issue. Feel free to edit it or add more information.